### PR TITLE
feat(route): output full text for msn news

### PR DESCRIPTION
**Pull request by KTachibanaM**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/msn/zh-cn/新华社/sr-vid-769jkard8p2sjcwjkv264h2divi25i5qk62whg7yq8akhwk23p6a?mode=fulltext
/msn/zh-cn/人民网/sr-vid-jwtjursmmc5egygacdnch06pxwhgdhk4in2r85yucvdgfgyhnsbs?mode=fulltext
/msn/zh-cn/澎湃新闻/sr-vid-kr7w4fq563568erb38y3ae3gr34vqe8vt4fr4sk574d69hu9h73s?mode=fulltext
/msn/zh-cn/红星新闻/sr-vid-ed4u3ky0u6b59hu83thk3pyy0fn6dfbnsyxsb79fvvsj34d0yk0s?mode=fulltext
/msn/zh-hk/AFP/sr-vid-3kv7h73mtcdhywg28d4f9ihgi4xcniq2ubb83iikdu3qwmbd73pa?mode=fulltext
/msn/zh-tw/Bloomberg/sr-vid-08gw7ky4u229xjsjvnf4n6n7v67gxm0pjmv9fr4y2x9jjmwcri4s?mode=fulltext
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
Please [cache](https://docs.rsshub.app/joinus/advanced/use-cache) the returned object

**Comment by KTachibanaM** [comment]
There still seems be to a bug. Please do not merge yet.

**Comment by KTachibanaM** [comment]
I have a weird issue locally where if I initiate a request to the msn route for the first time, the response has all valid content, but if I initiate the same request to the msn route for second or third time, the response has all empty values, e.g.

image

I couldn't replicate the issue with other routes, or the msn route that's already merged on master branch. @TonyRL do you have any idea why this might be happening?

File: lib/routes/msn/index.ts [hunk 1]
```diff
@@ -44,13 +45,36 @@ export const route: Route = {
...
+            let articleId = parsedArticleUrl?.pathname.split('/').pop();
+            if (articleId?.startsWith('ar-')) {
+                articleId = articleId.substring(3);
+                const fetchedArticleContentHtml = (await cache.tryGet(
+                    articleId,
+                    async () => {
+                        const articleData = await ofetch(`https://assets.msn.com/content/view/v2/Detail/${market}/${articleId}`);
+                        return articleData.body;
+                    },
+                    7_776_000
```
Comment (TonyRL) [inline lib/routes/msn/index.ts:62]: No need to hard-coded cache duration as it should be managed by the default value (config.cache.contentExpire)
```suggestion
```

**Comment by TonyRL** [comment]
Because you're using `mode` which causes RSSHub to clear and refetch whatever is stored in `description`

**Comment by KTachibanaM** [comment]
@TonyRL by `mode` are you referring to `mode=fulltext`? I think this issue still exists if I don't use `mode=fulltext` though.

**Comment by TonyRL** [comment]
Wrap `.map` with `Promise.all`

**Comment by KTachibanaM** [comment]
@TonyRL That seems to fix it. Thank you.

**Pull request merged**